### ISSUE TITLE
MultilineWhitespaceBeforeSemicolonsFixer chained calls with switch bug

### ISSUE THREAD

**Issue by zacharylund** [PHP-CS-Fixer/PHP-CS-Fixer#6985]
### Description

Semicolons for chained calls are not indented correctly when inside of a switch statement. This is a bug in the multiline_whitespace_before_semicolons rule when strategy is set to new_line_for_chained_calls. Enabling the statement_indentation rule fixes this, but only if the rule is enabled AFTER the multiline_whitespace_before_semicolons rule.

### Runtime version

```
PHP CS Fixer 3.17.0 Brazilian Kangaroo by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.0.28
```

### Used command

```
php-cs-fixer fix --dry-run -vvv --diff 
PHP CS Fixer 3.17.0 Brazilian Kangaroo by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.0.28
Loaded config default from "/tmp/php-cs-fixer/.php-cs-fixer.dist.php".
Using cache file ".php-cs-fixer.cache".
F                                                                                                            1 / 1 (100%)
Legend: .-no changes, F-fixed, S-skipped (cached or empty file), I-invalid file syntax (file ignored), E-error
   1) file.php (multiline_whitespace_before_semicolons)
      ---------- begin diff ----------
--- /tmp/php-cs-fixer/file.php
+++ /tmp/php-cs-fixer/file.php
@@ -5,6 +5,6 @@
         $object
             ->methodA()
             ->methodB()
-        ;
+    ;
         break;
 }

      ----------- end diff -----------


Found 1 of 1 files that can be fixed in 0.002 seconds, 12.000 MB memory used

```
### Configuration file

```php
<?php

return (new PhpCsFixer\Config())
    ->setRules([
        'multiline_whitespace_before_semicolons' => [
            'strategy' => 'new_line_for_chained_calls',
        ],
        //'statement_indentation' => true,
    ])
    ->setFinder(PhpCsFixer\Finder::create()->in(__DIR__))
;
```

### Code snippet that reproduces the problem

```php
<?php

switch ($object) {
    case 'foo':
        $object
            ->methodA()
            ->methodB()
        ;
        break;
} 
```

**Comment by zacharylund** [comment]
3.16.0 indents correctly:

```
php-cs-fixer fix --dry-run -vvv --diff 
PHP CS Fixer 3.16.0 BoY42 by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.0.28
Loaded config default from "/tmp/php-cs-fixer/.php-cs-fixer.dist.php".
Using cache file ".php-cs-fixer.cache".
F                                                                                                            1 / 1 (100%)
Legend: .-no changes, F-fixed, S-skipped (cached or empty file), I-invalid file syntax (file ignored), E-error
   1) file.php (multiline_whitespace_before_semicolons)
      ---------- begin diff ----------
--- /tmp/php-cs-fixer/file.php
+++ /tmp/php-cs-fixer/file.php
@@ -5,6 +5,6 @@
         $object
             ->methodA()
             ->methodB()
-    ;
+        ;
         break;
 }

      ----------- end diff -----------


Found 1 of 1 files that can be fixed in 0.002 seconds, 12.000 MB memory used
```

**Comment by Wirone** [comment]
Fixed in #6985 🙂
